src/protojson/UberJobReadyMsg.h

```diff
@@ -39,12 +39,55 @@
 // This header declarations
 namespace lsst::qserv::protojson {
 
+class UberJobStatusMsg {
+public:
+    using Ptr = std::shared_ptr<UberJobStatusMsg>;
+    virtual std::string cName(const char* fName) const { return std::string("UberJobStatusMsg") + fName; }
+    UberJobStatusMsg() = delete;
+    UberJobStatusMsg(UberJobStatusMsg const&) = delete;
+    UberJobStatusMsg& operator=(UberJobStatusMsg const&) = delete;
+    virtual ~UberJobStatusMsg() = default;
+
+    virtual std::shared_ptr<nlohmann::json> toJsonPtr() const = 0;
+
+    virtual bool equals(UberJobStatusMsg const& other) const = 0;
+
+    std::string const& getWorkerId() const { return _workerId; }
+    std::string const& getCzarName() const { return _czarName; }
+    CzarId getCzarId() const { return _czarId; }
+    QueryId getQueryId() const { return _queryId; }
+    UberJobId getUberJobId() const { return _uberJobId; }
+
+    /// Returns a string for logging.
+    virtual std::ostream& dumpOS(std::ostream& os) const;
```

Comment:
You don't have to name this function differently. C++ allows selective overloading if you define both methods as:
```
virtual std::ostream& dump(std::ostream& os) const;
std::string dump() const;
```